Sources/Danger/GitHubDSL.swift

```diff
@@ -17,489 +13,449 @@ public struct GitHub: Decodable {
+
+        /// The SHA for the commit.
+        public let sha: String
+
+        /// The raw commit metadata.
+        public let commit: GitCommit
+
+        /// The URL for the commit on GitHub.
+        public let url: String
+
+        /// The GitHub user who wrote the code.
+        public let author: User?
+
+        /// The GitHub user who shipped the code.
+        public let committer: User?
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+
+            sha = try container.decode(String.self, forKey: .sha)
+            commit = try container.decode(GitCommit.self, forKey: .commit)
+            url = try container.decode(String.self, forKey: .url)
+            author = (try? container.decodeIfPresent(User.self, forKey: .author)) ?? nil
+            committer = (try? container.decodeIfPresent(User.self, forKey: .committer)) ?? nil
+        }
     }
+}
 
-    // MARK: - Properties
-
-    /// The id number of the issue
-    public let id: Int
-
-    /// The number of the issue.
-    public let number: Int
-
-    /// The title of the issue.
-    public let title: String
-
-    /// The user who created the issue.
-    public let user: GitHubUser
+extension GitHub {
+    public struct Issue: Decodable, Equatable {
+        enum CodingKeys: String, CodingKey {
+            case id
+            case number
+            case title
+            case user
+            case state
+            case assignee
+            case assignees
+            case milestone
+            case body
+            case labels
+            case commentCount = "comments"
+            case isLocked = "locked"
+            case createdAt = "created_at"
+            case updatedAt = "updated_at"
+            case closedAt = "closed_at"
+        }
 
-    /// The state for the issue: open, closed, locked.
-    public let state: IssueState
+        public enum State: String, Decodable {
+            case open
+            case closed
+            case locked
+        }
 
-    /// A boolean indicating if the issue has been locked to contributors only.
-    public let isLocked: Bool
+        public struct Label: Decodable, Equatable {
+            /// The id number of this label.
+            public let id: Int
 
-    /// The markdown body message of the issue.
-    public let body: String?
+            /// The URL that links to this label.
+            public let url: String
 
-    /// The comment number of comments for the issue.
-    public let commentCount: Int
+            /// The name of the label.
+            public let name: String
 
-    /// The user who is assigned to the issue.
-    public let assignee: GitHubUser?
+            /// The color associated with this label.
+            public let color: String
+        }
 
-    /// The users who are assigned to the issue.
-    public let assignees: [GitHubUser]
+        /// The id number of the issue
+        public let id: Int
```

Comment:
<!--
  0 failure: 
  1 warning:  Variable name sho...
  
  
  DangerID: danger-id-Danger;
  File: Sources/Danger/GitHubDSL.swift;
  Line: 356;
-->

- :warning: Variable name should be between 3 and 40 characters long: 'id' (`identifier_name`)